app/src/org/commcare/activities/FormEntryActivity.java

```diff
@@ -1083,6 +1088,31 @@ protected void deliverError(FormEntryActivity receiver, Exception e) {
         }
     }
 
+    private String retrieveAndValidateSerializedFormIndex(int sessionDescriptorId) {
+        String interruptedFormIndex = HiddenPreferences.getInterruptedFormIndex();
+        if (interruptedFormIndex == null || interruptedFormIndex.isEmpty()){
+            return null;
+        }
+
+        int spaceIndex = interruptedFormIndex.indexOf(" ");
+        if (spaceIndex != -1) {
+            try {
+                int ssdId = Integer.parseInt(interruptedFormIndex.substring(0, spaceIndex));
+                String serializedFormIndex =
+                        interruptedFormIndex.substring(
+                                spaceIndex + 1, interruptedFormIndex.length());
+                if (ssdId == sessionDescriptorId) {
+                    return serializedFormIndex;
+                }
+            } catch (Exception e) {
+                // something went wrong
+            }
+        }
+        // data format is invalid, so better to clear the data
+        HiddenPreferences.clearInterruptedFormIndex();
```

Comment:
we will also reach here when `ssdId == sessionDescriptorId` eval to false which is not a case of invalid data format right ?

Reply:
Yeah, this is supposed to be short-lived, so if the `ssdIDs` don't match we can just clear it out, but happy to keep it there

Reply:
agree that it makes sense to clear it.